Dockerfile

```diff
@@ -1,7 +1,7 @@
 # use multi-stage builds to independently pull dependency versions
 ARG quilc_version=1.20.0
 ARG qvm_version=1.17.1
-ARG python_version=3.9
+ARG python_version=3.10
```

Comment:
Somebody who knows better than me should confirm that we can do this upgrade.

Reply:
I'm not sure how many people are installing using the dockerfile, but I think it should be fine to update this since the "breaking" release will just be at a new tag - we should probably use a `feat:` prefix on the commit though so this isn't just a patch release.

Looking at https://github.com/rigetti/pyquil/blob/master/.github/workflows/test.yml , we are still using `3.9` for testing - I think we should consider updating those as well (in a later mr)